### PR TITLE
Added compliant and noncompliant DL case for cloudformation

### PR DESCRIPTION
**Pull request by aastjais**
Add non compliant and compliant samples for cloudformation detectors.

Note: All the test cases have 100% recall and precision.

### CODE REVIEW

File: cloudformation/src/detectors/cloudformation-ec2-inst-detailed-monitorng/cloudformation-ec2-inst-detailed-monitorng_non-compliant.yaml [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+
+# {fact rule=cloudformation-ec2-inst-detailed-monitorng@v1.0 defects=1}
+Resources:
+  ExampleEC2:
+    Type: AWS::EC2::Instance
+    Properties:
+      ImageId: !Ref LatestAmiId
+      InstanceType: !Ref pInstanceType
+      IamInstanceProfile: !Ref rInstanceProfile
+      # Noncompliant: `Monitoring` is not enabled for your EC2 instances.
```
Comment (pratik-amazon) [inline cloudformation/src/detectors/cloudformation-ec2-inst-detailed-monitorng/cloudformation-ec2-inst-detailed-monitorng_non-compliant.yaml:12]: # Noncompliant: `Monitoring` is  disabled for EC2 instances.

File: cloudformation/src/detectors/cloudformation-eks-endpoint-no-public-access/cloudformation-eks-endpoint-no-public-access_compliant.yaml [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+# SPDX-License-Identifier: MIT-0
+
+# {fact rule=cloudformation-eks-endpoint-no-public-access@v1.0 defects=0}
+Resources:
+  ExampleEKSCluster:
+    Type: "AWS::EKS::Cluster"
+    Properties:
+      ResourcesVpcConfig:
+        SubnetIds: !Ref SubnetIds
+        # Compliant: `EndpointPublicAccess` is not `PubliclyAccessible`.
```
Comment (pratik-amazon) [inline cloudformation/src/detectors/cloudformation-eks-endpoint-no-public-access/cloudformation-eks-endpoint-no-public-access_compliant.yaml:11]: Correct this as per guideline shared.

File: cloudformation/src/detectors/cloudformation-eks-endpoint-no-public-access/cloudformation-eks-endpoint-no-public-access_non-compliant.yaml [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+# SPDX-License-Identifier: MIT-0
+
+# {fact rule=cloudformation-eks-endpoint-no-public-access@v1.0 defects=1}
+Resources:
+  ExampleEKSCluster:
+    Type: "AWS::EKS::Cluster"
+    Properties:
+      ResourcesVpcConfig:
+        SubnetIds: !Ref SubnetIds
+        # Noncompliant: `EndpointPublicAccess` is `PubliclyAccessible`.
```
Comment (pratik-amazon) [inline cloudformation/src/detectors/cloudformation-eks-endpoint-no-public-access/cloudformation-eks-endpoint-no-public-access_non-compliant.yaml:11]: Correct this as per guideline

File: cloudformation/src/detectors/cloudformation-restricted-ssh/cloudformation-restricted-ssh_compliant.yaml [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+  InstanceSecurityGroup:
+    Type: AWS::EC2::SecurityGroup
+    Properties:
+      GroupDescription: Enable SSH access and HTTP from the load balancer only
+      SecurityGroupIngress:
+      - Description: Allow SSH access from trusted IP
+        IpProtocol: tcp
+        FromPort: 22
+        ToPort: 22
+      # Compliant: `CidrIp` is more restricted.
```
Comment (pratik-amazon) [inline cloudformation/src/detectors/cloudformation-restricted-ssh/cloudformation-restricted-ssh_compliant.yaml:15]: "CidrIp is more restricted to 203.0.113.42/0"

File: cloudformation/src/detectors/cloudformation-restricted-ssh/cloudformation-restricted-ssh_non-compliant.yaml [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+Resources:
+  InstanceSecurityGroup:
+    Type: AWS::EC2::SecurityGroup
+    Properties:
+      GroupDescription: Enable SSH access and HTTP from the load balancer only
+      SecurityGroupIngress:
+      - Description: Allow SSH access from trusted IP
+        IpProtocol: tcp
+        FromPort: 22
+      # Noncompliant: `CidrIp` is less restricted.
```
Comment (pratik-amazon) [inline cloudformation/src/detectors/cloudformation-restricted-ssh/cloudformation-restricted-ssh_non-compliant.yaml:14]: "CidrIp is too permissive (0.0.0.0/0), allowing unrestricted access"

File: cloudformation/src/detectors/cloudformation-subnet-auto-assign-public-ip-disabled/cloudformation-subnet-auto-assign-public-ip-disabled_non-compliant.yaml [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+# {fact rule=cloudformation-subnet-auto-assign-public-ip-disabled@v1.0 defects=1}
+Resources:
+  ExampleSubnet:
+    Type: AWS::EC2::Subnet
+    Properties:
+      VpcId:
+        Ref: myVPC
+      CidrBlock: 10.0.0.0/24
+      AvailabilityZone: "us-east-1a"
+      # Noncompliant: `MapPublicIpOnLaunch` is assigned to a public IP addres.
```
Comment (pratik-amazon) [inline cloudformation/src/detectors/cloudformation-subnet-auto-assign-public-ip-disabled/cloudformation-subnet-auto-assign-public-ip-disabled_non-compliant.yaml:13]: correct the spelling of address

File: cloudformation/src/detectors/cloudformation-cloudfront-origin-failover-enabled/cloudformation-cloudfront-origin-failover-enabled_compliant.yaml [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+        ViewerCertificate:
+          AcmCertificateArn: "cert-test"
+          MinimumProtocolVersion: TLSv1.2_2021
+          SslSupportMethod: sni-only
+        WebACLId: arn:aws:wafv2:us-east-1:123456789012:global/webacl/ExampleWebACL/12345
+        Logging:
+          Bucket: myawslogbucket.s3.amazonaws.com
+        OriginGroups:
+          Quantity: 1
+          # Compliant: Origin Group contains two Origin Group Members.
```
Comment (pratik-amazon) [inline cloudformation/src/detectors/cloudformation-cloudfront-origin-failover-enabled/cloudformation-cloudfront-origin-failover-enabled_compliant.yaml:19]: "Origin Group includes two members"

File: cloudformation/src/detectors/cloudformation-cloudfront-origin-failover-enabled/cloudformation-cloudfront-origin-failover-enabled_non-compliant.yaml [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+        ViewerCertificate:
+          AcmCertificateArn: "cert-test"
+          MinimumProtocolVersion: TLSv1.2_2021
+          SslSupportMethod: sni-only
+        WebACLId: arn:aws:wafv2:us-east-1:123456789012:global/webacl/ExampleWebACL/12345
+        Logging:
+          Bucket: myawslogbucket.s3.amazonaws.com
+        OriginGroups:
+          Quantity: 1
+          # Noncompliant: Origin Group does not contain two Origin Group Members.
```
Comment (pratik-amazon) [inline cloudformation/src/detectors/cloudformation-cloudfront-origin-failover-enabled/cloudformation-cloudfront-origin-failover-enabled_non-compliant.yaml:19]: "Origin Group is missing the required members"

File: cloudformation/src/detectors/cloudformation-cloudfront-sni-enabled/cloudformation-cloudfront-sni-enabled_compliant.yaml [hunk 9]
```diff
@@ -0,0 +1,18 @@
...
+    Type: AWS::CloudFront::Distribution
+    Properties:
+      DistributionConfig:
+        Logging:
+          Bucket: myawslogbucket.s3.amazonaws.com
+        WebACLId: arn:aws:wafv2:us-east-1:123456789012:global/webacl/ExampleWebACL/12345
+        ViewerCertificate:
+          AcmCertificateArn: "cert-test"
+          MinimumProtocolVersion: TLSv1.2_2021
+          # Compliant: The value of `SslSupportMethod` is `sni-only`.
```
Comment (pratik-amazon) [inline cloudformation/src/detectors/cloudformation-cloudfront-sni-enabled/cloudformation-cloudfront-sni-enabled_compliant.yaml:16]: "SslSupportMethod is set to sni-only"

File: cloudformation/src/detectors/cloudformation-cloudfront-sni-enabled/cloudformation-cloudfront-sni-enabled_non-compliant.yaml [hunk 10]
```diff
@@ -0,0 +1,18 @@
...
+    Type: AWS::CloudFront::Distribution
+    Properties:
+      DistributionConfig:
+        Logging:
+          Bucket: myawslogbucket.s3.amazonaws.com
+        WebACLId: arn:aws:wafv2:us-east-1:123456789012:global/webacl/ExampleWebACL/12345
+        ViewerCertificate:
+          AcmCertificateArn: "cert-test"
+          MinimumProtocolVersion: TLSv1.2_2021
+          # Noncompliant: The value of `SslSupportMethod` is not `sni-only`.
```
Comment (pratik-amazon) [inline cloudformation/src/detectors/cloudformation-cloudfront-sni-enabled/cloudformation-cloudfront-sni-enabled_non-compliant.yaml:16]: "SslSupportMethod is set to vip"

**Review by pratik-amazon — COMMENTED** [review]
Please address the suggested changes

File: cloudformation/src/detectors/cloudformation-ec2-inst-detailed-monitorng/cloudformation-ec2-inst-detailed-monitorng_compliant.yaml [hunk 11]
```diff
@@ -0,0 +1,14 @@
...
+
+# {fact rule=cloudformation-ec2-inst-detailed-monitorng@v1.0 defects=0}
+Resources:
+  ExampleEC2:
+    Type: AWS::EC2::Instance
+    Properties:
+      ImageId: !Ref LatestAmiId
+      InstanceType: !Ref pInstanceType
+      IamInstanceProfile: !Ref rInstanceProfile
+      # Compliant: `Monitoring` is enabled for your EC2 instances.
```
Comment (pratik-amazon) [inline cloudformation/src/detectors/cloudformation-ec2-inst-detailed-monitorng/cloudformation-ec2-inst-detailed-monitorng_compliant.yaml:12]: This is still not adress

**Review by pratik-amazon — COMMENTED** [review]
Please address all suggested changes

**Review by atonaamz — APPROVED** [review]

**Pull request merged**